shared/chat/conversation/list/index.desktop.js

```diff
@@ -1,244 +1,269 @@
+  state = {
+    isLockedToBottom: true,
+    listRerender: 0,
+    selectedMessageKey: null,
   }
 
-  componentWillUnmount () {
-    // Stop any throttled/debounced functions
-    this._onScroll.cancel()
-    this._recomputeListDebounced.cancel()
-    this._onScrollSettled.cancel()
-  }
+  _onAction = (message: Constants.ServerMessage, event: any) => { throw new Error('_onAction Implemented in PopupEnabledList') }
+  _onShowEditor = (message: Constants.Message, event: any) => { throw new Error('_onShowEditor Implemented in PopupEnabledList') }
+  _onEditLastMessage = () => { throw new Error('_onEditLastMessage Implemented in PopupEnabledList') }
 
-  componentWillUpdate (nextProps: Props, nextState: State) {
-    // If a message has moved from pending to sent, tell the List to discard
-    // heights for it (which will re-render it and everything after it)
-    // TODO this doesn't work for things that take a bit to load (imgs)
-    if (this._toRemeasure.length) {
-      this._toRemeasure.forEach(item => {
-        this._cellCache.clearRowHeight(item)
-        if (this._listIsGood()) {
-          this._list.recomputeRowHeights(item)
-        }
-      })
-      this._toRemeasure = []
+  componentDidUpdate (prevProps: Props, prevState: State) {
+    // Force a rerender if we passed a row to scroll to. If it's kept around the virutal list gets confused so we only want it to render once basically
+    if (this._keepIdxVisible !== -1) {
+      this.setState({listRerender: this.state.listRerender + 1}) // eslint-disable-line react/no-did-update-set-state
+      this._keepIdxVisible = -1
     }
+    this._lastRowIdx = -1 // always reset this to be safe
 
-    if (this._shouldForceUpdateGrid) {
-      this._shouldForceUpdateGrid = false
-      if (this._listIsGood()) {
-        this._list.forceUpdateGrid()
-      }
+    if (this.props.editLastMessageCounter !== prevProps.editLastMessageCounter) {
+      this._onEditLastMessage()
     }
   }
 
-  _listIsGood () {
-    return this._list && this._list.Grid
-  }
-
-  componentDidUpdate (prevProps: Props, prevState: State) {
-    if ((this.props.selectedConversation !== prevProps.selectedConversation) ||
-        (this.state.messages !== prevState.messages)) {
-      this.state.isLockedToBottom && this._scrollToBottom()
+  componentWillReceiveProps (nextProps: Props) {
+    if (this.props.selectedConversation !== nextProps.selectedConversation ||
+      this.props.listScrollDownCounter !== nextProps.listScrollDownCounter) {
+      this.setState({isLockedToBottom: true})
     }
 
-    if (this.props.editLastMessageCounter !== prevProps.editLastMessageCounter) {
-      this.onEditLastMessage()
+    if (this.props.messageKeys.count() !== nextProps.messageKeys.count()) {
+      if (this.props.messageKeys.count() > 1 && this._lastRowIdx !== -1) {
+        const toFind = this.props.messageKeys.get(this._lastRowIdx)
+        this._keepIdxVisible = nextProps.messageKeys.indexOf(toFind)
```

Comment:
i'm explicitly trying to keep some stuff out of this.state to not cause it to cause rerenders